---
title: 'Reviewer Comments for Eye in a Disk: eyeIntegration human pan-eye and body transcriptome database version 1.0'
institute:
  - ogvfb: Ophthalmic Genetics and Visual Function Branch, National Eye Institute, National Institutes of Health
date: '2019-04-22'
output:
  word_document:
    reference_docx: word-styles-reference-01.docx
    fig_caption: yes
    pandoc_args:
      - '--lua-filter=scholarly-metadata.lua'
bibliography: eyeIntegration_v1_app_manuscript.bib
csl: investigative-ophthalmology-and-visual-science.csl
---

```
AE Comments to Author:
The manuscript "Eye in a Disk: eyeIntegration human pan-eye and body transcriptome database version 1.0" is an interesting compilation of information from many datasets. The reviewers have a number of comments and queries for consideration. 


Editorial Board Member (Comments for the Author (Required)):

This manuscript describes a database on genes from tissues of the eye into an accessible system for other researchers to access. They use one particular example, fetal eye tissue and organoid retina, as an example of the power of the database as constructed. While there is no hypothesis here, the ability of others to mine this curated data is likely to be of great interest to many researchers. Both reviewers have requested clarification and elaboration.

1. Figure 4 is not readable. If this is going to be useful illustration of potential, it would be better to do each of the 5 subparts as a separate figure. Similarly, Figure 5 would be better separated into 2 figures with increased readability.
2. Minor typos in the manuscript, but otherwise clearly written.
```

```
Reviewer #1 (Comments for the Author (Required)):

Building on the eyeIntegration web resource released in 2017, the authors re-wrote the bioinformatics pipeline and created one of the most comprehensive "reproducible and versioned transcriptome dataset" - eye in a Disk (EiaD) and updated the eyeIntegration web app, now integrating deep sequencing data from 371 human eye tissue samples and 848 GTEx samples across 44 tissues to allow quick search and visualization of human eye-related transcriptomes and make comparisons and identify differentially expressed genes among 14 various eye tissues and 44 other body tissues. It also included newly-published single cell seq data of developing and adult mouse retina. It is a powerful addition to the bioinformatics tools for the visual science community. In general, the manuscript is straightforward description of the development of the dataset and a guide to the web app. There are some minor clarifications should be done:
1) The authors quantified expression "across 42 gene and 46 transcript types" (supplemental table 2). Please describe the difference or relationship between of the "gene" and "transcript" types. Are they pretty much overlapping to each other?
2) "Figure 2: Raw RNA seq data ....... Visualizations across 63 tissues": are these 63 tissues the same as described in the abstract "44 body and 14 eye tissues", what are the extra 5 [63-(44+14)]? 
3) Line 179: "To create a high quality final dataset across the 1311 initial samples (suppl fig and table 2)". Are these "1311 initial samples" the same as "371 human eye tissue samples and 848 GTEx" described on line 158-159. If so, 371+848 â‰  1311? 
4) In Figure 3, the colors of ESC and lens, retina and RPE are too close to each other, and hard to be distinguished. More distinctive, different colors would help. What is the total number of clusters? A table listing all the clusters and the composition of each cluster should be included as a supplemental table. 
5) Line 208, ".... Then genes (or transcripts), then tissues." The "genes" should be "gene IDs by official gene name". The tables derived from "expression" search (e.g. Fig.4A) should be further annotated, e.g. define what are "decile", "log2(PMT+1)", etc. 
6) Since this dataset and web app is built in the way that "future sample updates can be streamlined in with less effort", perhaps advice on how future deep sequencing and single cell seq data should be organized for streamline updates should be added to the discussion.
7) Many typos and some unfished edit should be corrected, e.g. Line 285 "citation" should be added!



Reviewer #2 (Comments for the Author (Required)):

In this manuscript the authors describe aggregation of RNA-seq data from 371 ocular tissue samples into an "Eye in a Disk (EiaD)" database. Samples from adult and fetal retina, retinal pigment epithelium (RPE), cornea, lens, induced pluripotent stem cell (iPSC) derived retinal organoids are included. Data from two single cell RNA-seq (scRNA-seq) datasets generated from developing mouse retina are also included. The aggregated data are presented for general use via the eyeIntegration web interface. Data from 844 Genotype-Tissue Expression (GTEx) project samples derived from 44 different tissues are included for reference. To generate the EiaD database, data from RNA-seq studies that were available in the NIH Short Read Archive (SRA) were re-analyzed using the alignment free quantification tool Salmon, using the Gencode transcriptome index as a reference. Transcripts with low levels of expression (<2 TPM) were removed from the resulting dataset, as were samples with low calculated mapping rates. The resulting data were stored in an SQLite database. By using a workflow management system called SnakeMake, the authors ensure that reproducibility is maintained with each iteration of the data, facilitating sample tracking and updates of the database and portal, as new transcriptome data for the eye and other tissues become available. The whole analysis pipeline is available via Github, as is the R package used to generate the eyeIntegration web application. 

In addition to generating the EiaD, the authors created an interactive website with various useful features for visualizing and exploring expression data, such as differential gene expression between tissues, gene and tissue clustering based on expression profiles, weighted gene correlation networks around genes specified by the user, and t-SNE plotting, a data visualization tool for comparing expression data between different sample sets. To demonstrate these tools, the authors used t-SNE and PCA analyses to compare the gene expression profiles of the ocular and non-ocular tissues in the EiaD. They found that 
in general data from eye tissues cluster together, and that these eye tissues clusters are mostly separate from non-ocular GTEx tissue clusters. Further, analyses of RNA-seq data from iPSC derived retinal organoids and fetal retinal samples showed that the retinal organoids are similar but not identifical to fetal retina with regard to gene expression profiles. For example, HOXB genes are expressed in the retinal organoids, not fetal retina. 

The EiaD and eyeIntegration web application are a unique transcriptome database and data visualization portal centered on the eye, which enables quick exploration of gene and transcript expression between various healthy eye tissue types and a range of non-ocular human tissues. While the database and web tools are a valuable resource for the vision research community, there are a number of issues that need to be addressed/clarified.

Comments/Issues:

1. A common practice following identification of differential gene expression between two tissue types is experimental validation of a subset of the identified differences. Given the "dry lab" nature of this manuscript, this does not seem appropriate to request. But, it would be helpful to know if the differences detected in gene expression between iPSC derived retinal organoids and fetal retinal samples have been reported by other investigators. Similarly, it would be helpful to compare the output of the EiaD pipeline with similar data from the GTEx pipeline. For example, since the differences in gene expression between tissue types has been pre-calculated, it should be relatively easy to compare differential expression results for comparisons between GTEx tissues produced from GTEx portal and from the EiaD pipeline.

2. It would be helpful to provide some information about the accuracy of the Salmon transcript quantification tool for the reported analyses. For example, a search for Rhodopsin expression in the single cell RNA seq data shows relatively high expression in cone cells, in which rhodopsin is not usually expressed. Is this an artefact of the scRNA-seq method, or a possible read assignment error from the Salmon tool?

This question is also being asked because Salmon is an alignment-free method that performs light weight mapping of reads to transcript models and estimates expression levels of transcripts correcting for potential biases such as GC content. Given the limitations of transcript reconstruction from short RNA-seq reads, a commonly used approach to unify data from different studies, including GTEx, ENCODE, TopMed, and TCGA has been uniform realignment, gene expression quantification, and batch effect removal (e.g. Sci Data 2018 - DOI: 10.1038/sdata.2018.61). While Salmon has been reported to be among the top performing methods for transcript quantification, it also has limitations. For example, Salmon does not seem to perform as well for lowly expressed transcripts, and its efficacy can be affected by incompleteness of the gene/transcript annotation model or total number of isoforms per gene (BMC Genomics 18:583, 2017; Briefings in Bioinformatics 18: 260, 2017). It would be helpful to include information about the choice and use of this method in the manuscript.

3. It would be helfpul to learn more about how the authors handled non-biological batch effects between studies in their differential expression analyses. For example, in the pan tissue t-SNE plot shown in Figure 1 the majority of eye tissues cluster next to each other, but there are two separate clusters of retina and RPE samples that are distant from the other retina and RPE sample sets. Are these differences real, or due to differences in the RNA-seq library preparations and sequencing protocols between the different samples downloaded from SRA? One potential approach to this issue is to remove non-specific variation from the expression values in the differential expression model, as reported by Wang et al in their unification of GTEx and TCGA RNA-seq sample data in order to compare expression levels between normal and tumor samples from two different studies (Sci Data 2018, DOI: 10.1038/sdata.2018.61). One method that can be used to infer hidden covariates that capture much of the non-specific variation in expression datasets is ComBat in the R package SVAseq that enables indicating known variables that should be adjusted for, such as different studies (Nucleic Acids Res. 42:e161, 2014; Biostatistics 8:118-127, 2007). It is recommended that this be applied to each tissue type separately, combining samples from the same tissue type, such as retina, across different studies, and adding the study as an indicator in ComBat. 

4. Further information about the quality control and filtering applied to the RNA-seq data would be helpful to include in the manuscript. For example, how were the parameters for filtered by a mapping rate of 40% per sample and median read count >2 across all genes per sample selected? Also, basic statistics related to read mapping such as min, max, average, 25 quartile, 75 quartile of mapping should be included. 

5. To identify outlier samples the authors measured the distances of each sample from the centroid coordinates per cluster using a t-SNE plot. t-SNE is good for visualization but not for quantification of relative distance between samples and sample sets. The t-SNE retains probabilities not distances and hence the Euclidean distances between points on the 2-dimensional t-SNE plot are not informative. t-SNE documentation suggests that Kullback-Leibler divergences reported by t-SNE can be used for this purpose. This reviewer recommends use of coordinates from other dimensionality reduction methods such as Principal Coordinates Analysis (PCoA) or PCA, or a correlation-based statistic that compares the expression profile of each sample to the profiles of all samples in a given sample set (D statistic) to identify outlier samples, such as described in Nature Genetics 46: 430, 2014. 

6. It would be helpful to learn why the authors chose 892 samples from 44 tissues in GTEx, when there are over 11,000 RNA-seq sample available for 50 tissues in the current GTEx release v7 that is available on dbGaP. Also, a recent publication from other investigators at NEI described a large RNA-seq dataset from human retinal samples (Nat Genet. 51:606-610, 2019). Will these data be incorporated into the eyeIntegration database? If so, it would be helpful if this were mentioned in the text. If not, this additional study and the resources derived from it could be referred to.
```

