---
title: 'Eye in a Disk: eyeIntegration version 1.00'
author:
  - Vinay Swamy:
      institute: ogvfb
  - David McGaughey:
      institute:
        - ogvfb
      correspondence: "yes"
      email: mcgaugheyd@mail.nih.gov
institute:
  - ogvfb: Ophthalmic Genetics and Visual Function Branch, National Eye Institute, National Institutes of Health
date: "1/17/2019"
output:
  word_document:
    reference_docx: word-styles-reference-01.docx
    pandoc_args:
      - '--lua-filter=scholarly-metadata.lua'
      - '--lua-filter=author-info-blocks.lua'
bibliography: eyeIntegration_v1_app_manuscript.bib
---

# Abstract

# Introduction

  RNA sequencing (RNA-seq) has become a valuable tool for understanding the role of gene expression in biology.  RNA-seq provides an accurate estimate of the expression of all genes in a given biological sample. By examining gene expression in this concerted manner researchers glean new insight into the complex regulatory mechanisms that affect gene expression and better understand the effect disease has on these mechanisms. For example, researchers used RNA-seq to establish the effects of the gene NRL on the transcriptomic regulatory network of developing photoreceptors. (Swaroop paper). Some studies have generated large RNA-seq data sets to accurately determine gene expression specific to individual tissues in the body. However, such studies require enormous amounts of funding, due to the large cost of RNA-seq and large number of samples required for these studies. Some major institutions like the NIH have performed such studies and made the data publicly available. For example, the GTEX project generated a large amount of RNA-seq data in order to determine gene expression ins 55 Human tissues and published the results in an online web-app [@gtexconsortiumGenotypeTissueExpressionGTEx2013]. This data serves as a valuable tool for researchers across the world, but lacks relevance to eye research as GTEX failed to cover the eye. 
  Another solution to the problem of creating large, high quality RNA-seq data sets is manual curation of data generated by multiple studies. There is a vast amount of publicly available RNA-seq data in the Sequence Read Archive (SRA), where investigators across the world deposit their raw sequencing data. As of January 17th 2019, the SRA has 493,821 human RNA-seq samples. Creating a curated dataset from multiple data sets presents a difficult challenge. Samples must be carefully selected and labelled with the proper metadata, evaluated for quality, and normalized to account for technical variation between studies. While many fields of research have made such databases (Other curated database), a robust database does not exist in vision science research. Previous attempts to make such a database relied on outdated technology like microarrays (iSyte), or had too narrow of focus (TIGEM).  To remedy this problem, we have created Eyeintegration, the largest curated RNA-seq database for ocular research. We provide accurate information about gene expression in multiple human eye tissues using published studies from the SRA. Additionally, we incorporate murine single cell RNA-seq data to provide researchers access to information about retinal development. The database is built using a highly reproducible and robust pipeline that includes a rigorous quality control procedure and automated publishing of the website. The database contains both gene and transcript expression data for the retina, retinal pigmented epithelium (RPE), cornea, lens and 44 other human organs. We performed multiple differential expression analysis between tissues types and provide a wide range of data visualizations that can be accessed through an easy-to-use web-app.  The database can be found at Eyeintegration.nei.nih.gov


# Methods

# Results

# Conclusions

# Bibliography
