---
title: 'Eye in a Disk: eyeIntegration human pan-eye and body transcriptome database version 1.0'
author:
  - Vinay Swamy:
      institute: ogvfb
  - David McGaughey:
      institute:
        - ogvfb
      correspondence: "yes"
      email: mcgaugheyd@mail.nih.gov
institute:
  - ogvfb: Ophthalmic Genetics and Visual Function Branch, National Eye Institute, National Institutes of Health
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
  word_document:
    reference_docx: word-styles-reference-01.docx
    fig_caption: yes
    pandoc_args:
      - '--lua-filter=scholarly-metadata.lua'
      - '--lua-filter=author-info-blocks.lua'
bibliography: eyeIntegration_v1_app_manuscript.bib
csl: investigative-ophthalmology-and-visual-science.csl
abstract: "PURPOSE: To develop an easily accessible and reliable RNA-seq transcriptome database of healthy human eye tissues and a matching reactive web application to query gene expression in eye and body tissues. METHODS: We downloaded the raw sequnce data for 1,375 RNA-seq samples across 54 tissues in the GTEx project as a non-eye reference set. We then queried several public repositories to find all healthy, non-perturbed, human eye-related tissue RNA-seq samples. The 916 eye and 1,375 GTEx samples were sent into a Snakemake-based reproducible pipeline we wrote to quantify all known transcripts and genes, removes samples with poor sequence quality and mislabels, normalizes expression values across each tissue, performs 882 differential expression tests, calculates GO term enrichment, and outputs all as a single SQLite database file: the Eye in a Disk (EiaD) dataset. Furthermore, we rewrote the web application eyeIntegration (https://eyeIntegration.nei.nih.gov) to display EiaD. RESULTS: The new eyeIntegration portal provides quick visualization of human eye-related transcriptomes published to date by database version, gene/transcript, 19 eye tissues, and 54 body tissues. As a test of the value of this unified pan-eye dataset, we showed that fetal and organoid retina are highly similar at a pan-transcriptome level but display distinct differences in certain pathways and gene families like protocadherin and HOXB family members. CONCLUSION: The eyeIntegration v1.0 web app serves the pan-human eye and body transcriptome dataset, EiaD. This offers the eye community a powerful and quick means to test hypotheses on human gene and transcript expression across 54 body and 19 eye tissues."
keywords: "RNA-seq, lens, cornea, retina, RPE, Snakemake, web, app"
---

```{r Setup..., message=FALSE, warning=FALSE, include=FALSE}
#knitr::opts_chunk$set(fig.pos = 'p') # Places figures on their own pages
knitr::opts_chunk$set(out.width = '100%', dpi=300)

library(tidyverse)
library(citr)
library(cowplot)
library(ggrepel)
library(colorspace)
library(flextable)
library(captioner)
library(dbscan)
library(ggforce)
library(cowplot)
library(viridis)
library(rworldmap)

# setup caption-ing
fig_cap <- captioner("Figure")
supFig_cap <- captioner("Supplemental Figure")
tab_cap <- captioner("Table")
supTab_cap <- captioner("Supplemental Table")

# attach data pools
# source('src/data_pull.R')
load('data/data_pull.Rdata')
# vinny sra search
load('data/human_tx_studies.Rdata')
# source('src/tsne_and_pca_calcs.R')
load('data/dimRed_plot_data.Rdata')
# run 'src/viewing_stats.Rmd'
load('data/visitor_stats.Rdata')
# write supplementary file for metadata
write_csv(core_tight_2019, path = 'supplementary_materials/supplementary_files__metadata.csv')
# power calculations
load('data/power_data.Rdata')

core_tight_2019 <- core_tight_2019 %>% mutate(Origin = case_when( grepl('Transformed', Sub_Tissue, ignore.case = T) ~ 'Cell Line',
                                                                  grepl('Fetal', Sub_Tissue) ~ 'Fetal Tissue',
                                                                  grepl('Stem', Sub_Tissue) ~ 'Stem Cell',
                                                                  grepl('Organoid', Sub_Tissue) ~ 'Organoid',
                                                                  grepl('Cell Line', Sub_Tissue) ~ 'Cell Line',
                                                                  TRUE ~ 'Adult Tissue')) %>% 
  select(-run_accession) %>% 
  unique() 

# set up maps
visitorMap_cap <- supFig_cap(name='visitorMap', caption = paste('As of June 2019, eyeIntegration has had usage across', cities %>% unique() %>% length(), 'cities and', geo_locate$countries %>% unique() %>% length(), 'countries.'))

world <- map_data("world")
world <- world[world$region != "Antarctica",]

```

# Introduction

## RNA-seq is the predominant technology for deciphering transcriptomes

From anterior to posterior along the light trajectory, the human eye is composed of the cornea, lens, retina, retinal pigment epithelium (RPE), and choroid. The differentiation, maturation, and function of these tissues is mediated through spatial and temporal specific transcript and gene expression patterns, also known as the transcriptome. Today, RNA-sequencing (RNA-seq) is the predominant technology for quantifying the transcriptome. Analysis of the transcripts' expression across tissue, time, and perturbation allows researchers to decipher the genetic controls of eye development and function. To this end, a wide variety of human tissue sources have been used to assess gene function, including primary tissue (fetal and post-mortem), differentiated stem cells, immortalized cell lines, and most recently, organoids. These tissue types have been deeply sequenced across the cornea [@chenIdentificationNovelMolecular2013a; @chngHighThroughputGene2013a; @chungTranscriptomicProfilingPosterior2017; @fraustoTranscriptomicAnalysisCultured2016a; @kabzaCollagenSynthesisDisruption2017; @ouyangWNT7APAX6Define2014a; @songDirectedDifferentiationHuman2016a], lens [@hanWnt5aContributesDifferentiation2018], retina [@aldiriDynamicEpigeneticLandscape2017a; @farkasTranscriptomeAnalysesHuman2013b; @hoshinoMolecularAnatomyDeveloping2017; @kaewkhawTranscriptomeDynamicsDeveloping2015; @kaewkhawTreatmentParadigmsRetinal2016; @liComprehensiveAnalysisGene2014a; @mustafiTranscriptomeAnalysisReveals2016; @pinelliAtlasGeneExpression2016; @whitmoreTranscriptomicAnalysisNasal2014b], and RPE (choroid) [@darrowDeletionDXZ4Human2016a; @harenzaTranscriptomicProfiling392017; @huIdentificationMiRNASignatures2012b; @liComprehensiveAnalysisGene2014a; @nozawaSAFARegulatesInterphase2017; @obersteinCellularResponsesHuman2017; @pengEngineeringBloodRetinalBarrier2013; @radekeRestorationMesenchymalRetinal2015a; @sainiNicotinamideAmelioratesDisease2017; @samuelAppropriatelyDifferentiatedARPE192017; @santaguidaAneuploidyinducedCellularStresses2015b; @shaoSpontaneousGenerationNovel2017; @shihRestorationMesenchymalRPE2017; @smithRetinalPigmentEpithelial2017; @stevensonGiantinknockoutModelsReveal2017; @tresiniCoreSpliceosomeTarget2015; @whewaySiRNAbasedFunctionalGenomics2015; @whitmoreTranscriptomicAnalysisNasal2014b; @auCharacterizationLincRNAExpression2017]. 

## The GTEx gene expression web app lacks eye-specific tissues
The Genotype-Tissue Expression (GTEx) Project has generated RNA-seq data across dozens of post-mortem human tissues from hundreds of unique donors and presents the gene and transcript level data in a comprehensive and user-friendly web app (https://gtexportal.org/); however eye tissues have not been included [@carithersNovelApproachHighQuality2015; @gtexconsortiumGeneticEffectsGene2017]. Recently Ratnapriya et al. have published a huge set of post-mortem retina, both normal and with varying degrees of age-related macular degeneration (AMD) and the GTEx project is providing the data as a download link. This data, as of June 2019, is not available in the interactive GTEx visualizations [@ratnapriyaRetinalTranscriptomeEQTL2019]. The Sequence Read Archive (SRA) and European Nucleotide Archive (ENA) are the primary repositories for all raw sequence data and two groups have quantified large portions of the RNA-seq data, including some human eye tissues, from the SRA: recount2 and ARCHS4 [@collado-torresReproducibleRNAseqAnalysis2017; @lachmannMassiveMiningPublicly2018]. To date, no curation of the sample level metadata has been done, therefore it is challenging to parse out which eye tissues are present and even more difficult to determine whether any samples were chemically or genetically perturbed. More targeted web resources that allow researchers to quickly assess gene expression in eye tissues include iSYTE, EXPRESS, and retina.Tigem.it [@budakExpressDatabaseTranscriptome2018a; @kakranaISyTEDatabaseExpressionbased2018; @pinelliAtlasGeneExpression2016]. However iSYTE only includes lens samples, EXPRESS is limited to a subset of mouse lens and retina samples, and retina.Tigem.it is retina only. We thus aimed our efforts at developing an easily accessible and reliable RNA-seq based transcriptome database of healthy human eye tissues and a matching reactive web application to query gene expression in eye and body tissues.

## The eyeIntegration app interactively serves huge GTEx and human eye tissue datasets (EiaD)
The eyeIntegration web resource (https://eyeIntegration.nei.nih), originally released in 2017 at version 0.6, provides the largest set of transcriptomes from hand-curated human eye tissues along with hundreds of GTEx tissue samples [@Bryan_2018]. This interactive web app allows for quick transcript and gene comparisons across many eye tissues and dozens of other body tissues. The dataset that the original eyeIntegration web app served was created with a series of scripts, several of which were run interactively to manually assess quality control for the samples. The interactive nature of some of the steps precluded efficient and regular data updates for the data.

To better meet the needs of the eye research community we have re-written the bioinformatic pipeline that creates the eye and body RNA-seq dataset to allow for regular, versioned updates for eyeIntegration. We call this reproducible and versioned transcriptome dataset "Eye in a Disk" (EiaD). The pipeline automates the EiaD creation, ensures full reproducibility of the results, allow for external data comparison, provides consistent sample quality control, and improves efficiency for future sample updates. The 2019 EiaD dataset contains several new tissue types, full gene product quantification, along with hundreds of new samples and improved sample labeling. The eyeIntegration web app has also been re-written to provide many new features, including versioned EiaD datasets, custom URL shortcut creation, new visualizations, improved data table searching, easy download of core datasets, and local install of the entire interactive resource with three commands. Additionally, we are prototyping new tools to display single cell RNA-seq (scRNA-seq) data to provide researchers access to cell type specific information about gene expression across murine retinal development.

## The EiaD dataset can be used to identify potential avenues to improve retina organoid maturation
Retina organoids are an increasingly popular means to model human retina development. We used our pan-study EiaD dataset to show that, at a pan-transcriptome level, organoids are highly similar to early fetal retina tissue. We also show that important temporal gene expression patterns in the fetal retina tissue are recapitulated in the organoids. As the organoid differentiation methods do not yet produce fully mature retina, we focused on identifying differentially expressed processes between organoid retina and embryonic retina and detected, for example, identifying protocadherin and HOXB family gene expression differences which suggest targetable pathways to improve and benchmark organoid differentiation methods.

# Methods
## Identification of potential eye samples 
We exhaustively searched the SRA with the SRAdb R package for eye related tissues using the query ‘cornea|retina|RPE|macula|fovea|choroid|sclera|iris|lens|eye’ across all columns and rows in the 'SRA' table [@leinonenSequenceReadArchive2011; @zhuSRAdbQueryUse2013a]. As the SRAdb is being deprecated we also ran searches on the SRA and Gene Expression Omnibus (GEO) web pages with as follows: ((“Homo sapiens"[orgn:__txid9606])  AND (transcriptomic[Source]) AND ("2019/01/01"[Publication Date] : "3000"[Publication Date]) AND (retina[Text Word] OR RPE[Text Word] OR macula[Text Word] OR fovea[Text Word] OR choroid[Text Word] OR sclera[Text Word] OR iris[Text Word] OR lens[Text Word] OR cornea[Text Word] OR ‘trabecular meshwork’[Text Word] OR ‘canals of schlemm’[Text Word] OR ‘cillary body’[Text Word] OR ‘optic nerve’[Text Word] OR ‘laminar cribosa’[Text Word] OR retina[Title] OR RPE[Title] OR macula[Title] OR fovea[Title] OR choroid[Title] OR sclera[Title] OR iris[Title] OR lens[Title] OR cornea[Title] OR ‘trabecular meshwork’[Title] OR ‘canals of schlemm’[Title] OR ‘cillary body’[Title] OR ‘optic nerve’[Title] OR ‘laminar cribosa’[Title] ))". We hand selected relevant studies and selected healthy, control or unmodified samples spanning primary adult tissue, primary fetal tissue, induced pluripotent stem cell (iPSC)-derived tissue, stem cell derived organoids, and immortalized cell lines. In order to compare gene expression in the eye against expression in other body tissues, we obtained samples from `r core_tight_2019 %>% filter(study_accession == 'SRP012682') %>% pull(Sub_Tissue) %>% unique() %>% length()` different body tissues from the GTEx project. Using SRA metadata from each study we extracted sample and run accessions, library type, tissue of origin, and sub-tissue of origin. Any of the preceding information missing from the SRA metadata was added by hand, when available. Stem cell-derived tissues and cell lines are marked as sub-tissues of the tissue they model.

## Raw data download and quantification
We downloaded the relevant SRA files for each sample directly from the NCBI ftp server using the file transfer software Aspera. SRA files were converted to FASTQ format using the tool fastq-dump from the SRAtoolkit software package [@leinonenSequenceReadArchive2011]. Samples only available in the BAM format were converted to FASTQ format using SAMTools [@liSequenceAlignmentMap2009]. Sample transcriptomes were quantified using the alignment free quantification software Salmon, using transcriptomic index built from gencode v28 protein coding transcript sequences using the transcriptomic aligner Salmon [@harrowGENCODEReferenceHuman2012; @patroSalmonProvidesFast2017a]. Using the resulting expression quantification, we identified lowly or unused transcripts within the gencode annotation, and removed transcripts that accounted for 5% or less of the total expression for its parent gene as per Sonneson et al [@sonesonIsoformPrefilteringImproves2016]. Samples were re-quantified against a transcriptomic index built on the filtered transcript sequences. The Salmon count values were quantified as (transcript) length scaled Transcripts Per Million (TPM) to the transcript and gene level using tximport [@sonesonDifferentialAnalysesRNAseq2016a].  

## Quality control 
We first removed samples with a Salmon calculated mapping rate less than 40%. This value was selected as being the far left tail of the distribution of mapping rates across samples (`r supFig_cap(name = 'mapping_rate_cap', display = 'cite')`). We removed lowly expressed genes by calculating the median expression across all samples for each gene and kept genes that had a median count >200 across all samples. To reduce the noise from experimental variability between each study, we normalized samples by sequence library size using the calcNormFactors function from the edgeR R package, and then quantile smoothed expression data using the R package qsmooth at the tissue level [@hicksSmoothQuantileNormalization2018a; @robinsonEdgeRBioconductorPackage2010]. In a change from our previous eyeIntegration work [@Bryan_2018], we now correct our counts for mapping rate and tissue type with the limma batchEffects function [@ritchieLimmaPowersDifferential2015]. The transformed values are used for the box plot and t-SNE visualizations. 

To identify outliers we followed an approach similar to a method in Wright et al [@wrightHeritabilityGenomicsGene2014]. Briefly, we first selected the 3000 genes with the highest variance across all samples and then for each sub-tissue type $T$ and each sample $i$ in $T$, we first calculated $r_{i}$, the average correlation between $i$ and all other samples in $T$. Next, we calculated $D_{i}$, where $D_{i} = \frac{(r_{i} - r\bar{\bar{}})}{median(r_{i} - r\bar{\bar{}})}$ and  $r\bar{\bar{}}$ is the grand mean of all $r_{i}$ for $i$ in $T$. We removed samples with $D_{i} <$ -17.5; we determined this threshold by generating a tSNE plot of our samples, and visually identifying outliers in adult retina tissue.  The ($max(D_{i})$) amongst these outliers was -17.58 and from this we chose -17.5 as our outlier threshold. 

## Differential Gene Expression Analysis and GO term enrichment
We used the non-transformed length scaled TPM values to determine differential gene expression between different sub-tissue types. First, we generated a synthetic body set to serve as single representative sub-tissue type for pan-body gene expression by randomly sampling GTEx tissues.  We used the voom function from the limma R package to convert gene expression to precision weights, and then performed pairwise differential expression tests for all combinations of eye sub-tissues (using mapping rate as a covariate), the synthetic body tissue, and human body tissues using an empirical Bayes test [@ritchieLimmaPowersDifferential2015; @lawVoomPrecisionWeights2014]. We extracted significant genes (FDR p < 0.01) for all `r limma_DE_tests %>% nrow()` comparisons and used these to calculate GO enrichment. The significant gene list for each eye sub-tissue was split into upregulated and down regulated sets and each set was tested for enrichment using the enrichGO function from the clusterProfiler R package (q-value < 0.01) [@yuClusterProfilerPackageComparing2012].

## eyeIntegration web app and R package
The data generated in the above steps is consolidated into a SQLite database, with the original dataset for eyeIntegration and the new 2019 EiaD dataset each getting a separate database file. The code that creates the eyeIntegration web app is written in Shiny and R and has been wrapped into an R package (https://github.com/davemcg/eyeIntegration_app/) that can be deployed on a local computer or a web server (https://eyeIntegration.nei.nih.gov). The app can be deployed on a local computer with 50GB of free disk space by running three commands in R: "devtools::install_github('davemcg/eyeIntegration_app')", "eyeIntegrationApp::get_eyeIntegration_datasets()", and  "eyeIntegrationApp::run_eyeIntegration()". 

## Snakemake reproducible pipeline
While the sample search and metadata parsing in a semi-curated process, the processing from the raw data to the creation of the SQLite EiaD database underlying eyeIntegration is wrapped in a Snakemake pipeline, which ensures full reproducibility of the results [@kosterSnakemakeScalableBioinformatics2012]. We make the code for the pipeline available at https://github.com/davemcg/EiaD_build. 

## scRNA-seq processing
The eyeIntegration site, as of June 2019, hosts two large scRNA-seq datasets from Macosko et al. and Clark et al [@macoskoHighlyParallelGenomewide2015a; @clarkSingleCellRNASeqAnalysis2019]. We use the processed gene count data directly from each group, as well as their cluster assignments which specify what cell type each individual cell is. The count data is mean averaged to the cell type, age, and gene level for the single cell expression section of eyeIntegration. We also display t-SNE and UMAP-based two-dimensional visualizations of the Macosko and Clark data, respectively, in the web app. For detail so the t-SNE processing we did on the Macosko dataset, see the methods of Bryan et al [@Bryan_2018]. 

## Power Calculation
We use the ssizeRNA R package to calculate power (p) across samples (n) at an FDR of 0.05 [@biSampleSizeCalculation2016]. Important parameters for ssize RNA include the variability (dispersions for the samples and genes), which were calculated directly from our EiaD length scaled TPM values by the edgeR packages estimateCommonDisp and estimateTagwiseDisp. The code to calculate the power is given as 'power_calc.R'.

## Manuscript as code and reproducibility
This manuscript's figures, tables, and most numbers, are all created and laid out in a R markdown document that interweaves code and text. The knitr and pandoc program is used to lay out the figures and tables and output a docx file. The code that generates this manuscript can be found at https://github.com/davemcg/eyeIntegration_v1_app_manuscript. 

The relevant code-bases (https://github.com/davemcg/eyeIntegration_v1_app_manuscript, https://github.com/davemcg/EiaD_build) and the EiaD dataset itself has been deposited into Zenodo with accession 10.5281/zenodo.3238677 to ensure the data can be accessed in the future, even should eyeIntegration and GitHub become inaccessible in the future. 

# Results

## EiaD 2019 contains `r ((core_tight_2019 %>% filter(Kept == 'Kept', study_accession != 'SRP151763') %>% pull(study_accession) %>% unique()) %in% (core_tight_2017$study_accession %>% unique())) %>% table() %>% as_tibble() %>% slice(1) %>% pull(n)` new human eye RNA-seq studies, `r (core_tight_2019 %>% filter(Kept == 'Kept', study_accession == 'SRP151763') %>% pull(sample_accession)) %>% length()` new Retina AMD samples, `r ((core_tight_2019 %>% filter(Kept=='Kept', study_accession != 'SRP012682', study_accession != 'SRP151763') %>% pull(sample_accession)) %>% length()) - core_tight_2017 %>% filter(study_accession != 'SRP012682') %>% nrow()` new eye samples, and `r ((core_tight_2019 %>% filter(Kept=='Kept', study_accession != 'SRP012682', study_accession != 'SRP151763') %>% pull(Sub_Tissue)) %>% unique() %>% length())`  total eye sub-tissue types

```{r, echo = F}
# set color levels
color <- c('#7fc97f','#beaed4','#ffff99','#fdc086','#386cb0','#f0027f', '#000000', '#000000', '#000000', '#000000', '#000000')
names(color) <- c('Cornea', 'ESC', 'Lens', 'Retina', 'RPE', 'GTEx', 'Brain', 'Pituitary', 'Blood', 'Skin', 'Bone Marrow')
custom_fill <- scale_fill_manual(values = color)
custom_col <- scale_colour_manual(values = color)
```

```{r tissueCountsBarPlot, fig.height = 5, fig.width = 6, fig.cap=tissueCountsBarPlot_cap, echo=F}
tissueCountsBarPlot_cap <- fig_cap(name='tissueCountsBarPlot', caption =  paste0('Substantial increase in eye tissue count and type from 2017 (', core_tight_2017 %>% filter(study_accession != 'SRP012682') %>% nrow(), ', lighter color) to 2019 (', core_tight_2019 %>% filter(Kept=="Kept", study_accession != 'SRP012682') %>% nrow(),', darker color) EiaD. We also improved the metadata labelling, the cornea samples (green) now delineates endothelial and epithelial tissues and the retina samples (orange) distinguish retina organoid and retinal ganglion cell (RGC) from stem cells. Counts for each bar plot given in the boxes. The y-axis is a log2 transformed count of samples passing our QC filters. '))

core_both <- bind_rows(core_tight_2019 %>% mutate(Version = '2019 EiaD') %>% filter(Kept == 'Kept'), core_tight_2017 %>% mutate(Version = '2017') %>% mutate(Sub_Tissue = gsub('_',' - ', Sub_Tissue))) %>% unique()


core_both  %>% mutate(Tissue = case_when(study_accession == 'SRP012682' ~ 'GTEx', TRUE ~ Tissue),
                      Sub_Tissue = case_when(study_accession == 'SRP012682' ~ 'GTEx', TRUE ~ Sub_Tissue),
                      Sub_Tissue = case_when(grepl('MGS', Sub_Tissue) ~ 'Retina - Adult AMD Tissue', TRUE ~ Sub_Tissue)) %>% 
  filter(grepl('GTEx|Retina|Cornea|Lens|RPE|ESC', Tissue)) %>% unique() %>% 
  #mutate(Sub_Tissue = case_when(sample_accession == 'SRS1955479' ~ 'Retina - Adult Tissue', TRUE ~ Sub_Tissue)) %>%
  group_by(Tissue, Sub_Tissue, Version) %>% summarise(Count=n()) %>% 
  mutate(Dataset = Version) %>% 
  ggplot(aes(x=Sub_Tissue, y= log2(Count), fill = Tissue, alpha = Dataset, label = Count, group = Dataset)) + 
  geom_bar(stat = 'identity', position = position_dodge(width = NULL)) + 
  theme_minimal() + 
  theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.3))  + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  xlab('') + 
  geom_label_repel(stat = 'identity', position = position_dodge(width = 0.9), size = 4, ylim =c(3, NA), show.legend = FALSE) + 
  ylab('log2(Count of Samples)') + scale_alpha_manual(values = c(0.5,1)) +
  #coord_cartesian(ylim = c(0,11)) +
  theme(text = element_text(size = 10)) +
  scale_y_continuous(breaks = c(0,3,6,9)) +
  custom_fill

```

Our query on May 8th 2019 to the SRA found `r human_tx_studies$study_accession %>% unique() %>% length()` potentially relevant studies. We removed non-pertinent studies and selected healthy or unmodified tissue from each relevant study for a total, including  of `r core_tight_2019$study_accession %>% unique() %>% length()` studies, `r ((core_tight_2019$study_accession %>% unique()) %in% (core_tight_2017$study_accession %>% unique())) %>% table() %>% as_tibble() %>% slice(1) %>% pull(n)` of which are new to the 2019 EiaD dataset. The 2019 EiaD dataset contains `r core_tight_2019 %>% filter(Kept == 'Kept') %>% filter(study_accession != 'SRP012682') %>% nrow()` human eye tissue samples and also includes `r core_tight_2019 %>% filter(Kept == 'Kept') %>% filter(study_accession == 'SRP012682') %>% nrow()` GTEx samples across `r core_tight_2019 %>% filter(Kept == 'Kept') %>% filter(study_accession == 'SRP012682') %>% pull(Sub_Tissue) %>% table() %>% length()` tissues for easy comparison (`r tab_cap(name = 'tableSampleCounts', display = 'cite')`, `r supTab_cap(name = 'tableSampleCountsGTEx', display = 'cite')`). The 2019 EiaD contains `r core_tight_2019 %>% filter(Kept == 'Kept', Tissue == 'ESC') %>% nrow()` undifferentiated iPSC, `r core_tight_2019 %>% filter(Kept == 'Kept', Tissue == 'Cornea') %>% nrow()` cornea, `r core_tight_2019 %>% filter(Kept == 'Kept', Tissue == 'Lens') %>% nrow()` lens, `r core_tight_2019 %>% filter(Kept == 'Kept', Tissue == 'Retina') %>% nrow()` retina, and `r core_tight_2019 %>% filter(Kept == 'Kept', Tissue == 'RPE') %>% nrow()` RPE (choroid) samples; in total we have added `r ((core_tight_2019 %>% filter(Kept=='Kept', study_accession != 'SRP012682') %>% pull(sample_accession)) %>% length()) - core_tight_2017 %>% filter(study_accession != 'SRP012682') %>% nrow()` new samples to the 2019 EiaD  (`r fig_cap(name = 'tissueCountsBarPlot', display = 'cite')`). We refer to native-tissue extracted RPE as RPE (choroid) because it is not possible to remove the choroid from the RPE without culturing. 

Stem cell-derived cornea, stem cell-derived lens, and fetal retina are three new types of sub-tissues that are now available in EiaD. We have also substantially improved the granularity of the cornea tissue metadata, now delineating whether the tissue is from the endothelium or epithelium (`r fig_cap(name = 'tissueCountsBarPlot', display = 'cite')`); previously these had been grouped together as adult tissue. Another substantial addition to the 2019 EiaD are non-protein coding genes; while protein-coding is the most common gene and transcript typse, there are dozens of different non-coding classes. The 2017 version of eyeIntegration only quantified protein coding genes and transcripts. We now quantify expression across `r gene_anno %>% count(gene_type) %>% as_tibble() %>% nrow()` gene and  `r tx_anno%>% count(transcript_type) %>% as_tibble() %>% nrow()` transcript types, including protein coding, retained intron, lincRNA, antisense, and pseuodogenes (`r supTab_cap(name = 'tx_and_gene_types', display = 'cite')`).

We have also added the large retina AMD post-morten Ratnapriya et al. cohort to EiaD 2019 [@ratnapriyaRetinalTranscriptomeEQTL2019]. This cohort contains hundreds of samples ranging from non-AMD (Minnesota Grading System (MGS) 1) to severe AMD (MGS 4). While eyeIntegration is intended to be a source for normal tissues, we have made an exception for this study, as this is a large cohort and AMD is a common disease. We found our corrections methods did not group the non-AMD Ratnapriya et al. samples with our other collected retina samples (see Retina MGS in `r supFig_cap(name = 'tSNE_sup', display = 'cite')`). This may be related to the lower mapping rate of the Ratnapriya et al. data (see Retina MGS in `r supFig_cap(name = 'mapping_rate_cap', display = 'cite')`).

## `r (core_tight_2019 %>% filter(Kept == 'Kept') %>% filter(study_accession == 'SRP012682') %>% pull(Sub_Tissue) %>% length()) - (core_tight_2017 %>% filter(study_accession == 'SRP012682') %>% pull(Sub_Tissue) %>% length())` more GTEx samples and `r (core_tight_2019 %>% filter(Kept == 'Kept') %>% filter(study_accession == 'SRP012682') %>% pull(Sub_Tissue) %>% table() %>% length()) - (core_tight_2017 %>% filter(study_accession == 'SRP012682') %>% pull(Sub_Tissue) %>% table() %>% length()) - 1` new GTEx body sub-tissue types added to 2019 EiaD
```{r echo = FALSE}
powerCurve_cap <- supFig_cap(name='powerCurve', caption = 'Power curve to assess ability to detect >= 1 log2(FoldChange) in gene expression between n samples (x-axis) in each group.')
```
Our previous dataset for eyeIntegration version 0.6 held about 20 samples per GTEx tissue type. We ran power calculations to assess our ability to detected >= 1 log2(Fold Change) in gene expression between two conditions to determine whether this is a sufficient number of samples (`r supFig_cap(name='powerCurve', display = 'cite')`). Our calculations suggest, for example, that we have 83% power to detect a 1 log2(Fold Change) difference in gene expression with two groups of twenty samples. To increase our power to make significant eye to body comparisons, we added about 10 more samples per GTEx tissue types (which at 30 samples, would give about 90% power). We also took this opportunity to add bladder, bone marrow, cervix uteri, fallopian tube, ovary, prostrate, testis, uterus, and vagina GTEx tissue samples (`r supTab_cap(name = 'tableSampleCountsGTEx', display = 'cite')`). 

## Rigorous quality control and reproducible workflow system ensures high quality transcriptomes that consistently cluster together by tissue type
```{bash, echo = F, eval = F}
# ran in biowulf2.nih.gov:/data/mcgaugheyd/projects/nei/mcgaughey/auto_eyeIntegration
# 1314 from running: for i in RE_quant_files/*/lib_format_counts.json; do echo $i; done | wc -l 
# 46030412878 from running: for i in RE_quant_files/*/lib_format_counts.json; do grep num_compatible_fragments $i; done |  cut -f2 -d":" | cut -f1 -d"," | awk '{sum += $1} END {print sum}'
```


```{r workflowDoodle, echo = F, fig.width = 10, fig.height = 10, results='asis', fig.cap = workflowDoodle_cap}
workflowDoodle_cap <- fig_cap(name='workflowDoodle', caption = paste0('Raw RNA-seq data from the SRA is run through our pipeline to create the EiaD, which is used by eyeIntegration app to serve interactive gene expression visualizations across ', core_tight_2019 %>% filter(Kept == 'Kept') %>% pull(Sub_Tissue) %>% unique() %>% length() - 1, ' tissues'))

knitr::include_graphics('figures_and_tables/simplified_workflow.png')
```

We built an automated pipeline for processing and analyzing all data for the web app using the program Snakemake, a python-based workflow management system that allows for efficient parallel execution of the analysis, facilitates reproduction by others, and simplifies long term maintenance of the EiaD data (`r fig_cap(name='workflowDoodle', display = 'cite')`, `r supFig_cap(name='snakemakeDAG', display = 'cite')`)[@kosterSnakemakeScalableBioinformatics2012]. To create a high quality final dataset across the `r core_tight_2019 %>% nrow()` initial samples (`r supTab_cap(name = 'metadataTable', display = 'cite')`) and 67,315,523,736 reads we developed a rigorous quality control procedure as part of our analysis, considering a sample’s read mapping rate and median count level as well as behavior relative to samples of the same sub-tissue type (see Methods).  To identify tube mislabeling or sample extraction issues, we used sample-level gene correlation metrics (see methods) to identify variability within samples of the same sub-tissue and ensure overall consistency in data processing (`r fig_cap(name = 'tSNE', display = 'cite')`). After these steps `r core_tight_2019 %>% filter(Kept != 'Kept') %>% filter(study_accession != 'SRP012682') %>% nrow()` eye samples and `r core_tight_2019 %>% filter(Kept != 'Kept') %>% filter(study_accession == 'SRP012682') %>% nrow()` GTEx samples were removed. 

After our quality control and processing workflow, we found that samples of the same tissue type and origin cluster well together (`r fig_cap(name = 'tSNE', display = 'cite')` and `r supTab_cap(name = 'tSNE_table', display = 'cite')`)). For example, in the retina group, primary adult tissue clusters tightly and distinctly from other cell types, and retinal organoids and fetal retina samples cluster together. Our ability to uniformly cluster data by known biological source independent of study origin demonstrates that our workflow can effectively account for technical variation between studies. 

While t-SNE is a powerful algorithm for grouping samples, it is not consistent for determining the relationships between clusters [@wattenbergHowUseTSNE2016a]; PCA is more useful in this regard. We ran a PCA dimensionality reduction (`r supFig_cap(name = 'allSamplePCA', display = 'cite')`) on all samples, finding that the eye tissues still generally group together and apart from all other human body tissues. Adult retina is most similar to the brain tissue. RPE and cornea are most similar to blood, bone marrow, and skin. 

```{r tsne, fig.height = 15, fig.width = 20, echo = F, message = F, fig.cap=tsne_cap}
tsne_cap <- fig_cap(name='tSNE', caption = 't-SNE two-dimensional transcriptome profiles by sample demonstrate effective quality control and transcriptome processing. Colors match different tissue types and shapes of points define the origin of the tissues.')

tsne_50 <- tsne_50 %>% filter(Tissue != 'Choroid Plexus', !grepl('MGS', Sub_Tissue))
dbscan_cluster <- dbscan(tsne_50 %>% select(X1, X2),  minPts = 3, eps = 2.5)
tsne_50$Cluster <- dbscan_cluster$cluster
col1 = 'X1'
col2 = 'X2'
# create label points for each cluster
cluster_centers <- tsne_50 %>%
  left_join(.,core_tight_2019) %>% group_by(Cluster) %>%
  summarise(C1=mean(!!sym(col1)),C2=mean(!!sym(col2)),Tissue=paste(unique(Tissue),collapse=','))

# samples closest to center in each cluster
center_samples <- tsne_50 %>% left_join(.,core_tight_2019)  %>%
  left_join(.,cluster_centers, by=c('Cluster')) %>%
  mutate(Distance = (!!sym(col1)-C1)+(!!sym(col2)-C2)) %>%
  group_by(Cluster) %>%
  dplyr::slice(which.min(Distance)) %>%
  dplyr::slice(1) %>% 
  .[['sample_accession']]

# cluster stats
cluster_stats <- tsne_50 %>% left_join(.,core_tight_2019)  %>%
  mutate(Origin = gsub('Organoid','Stem Cell', Origin)) %>% 
  mutate(Origin=factor(Origin, levels=c('Adult Tissue', 'Fetal Tissue', 'Stem Cell', 'Cell Line'))) %>%
  mutate(Cluster = as.factor(Cluster)) %>%
  group_by(Cluster) %>%
  summarise(Cluster_Tissues = paste(unique(Tissue), collapse=',\n'), Cluster_Counts = paste(n(), ' samples', sep=''))

# set up for ggplot
tsne_50_prep <- tsne_50 %>% select(-Origin) %>% 
  left_join(., core_tight_2019 %>% select(sample_accession, Origin), by = 'sample_accession') %>% 
  mutate(Origin=factor(Origin, levels=c('Adult Tissue', 'Fetal Tissue', 'Stem Cell', 'Cell Line'))) %>%
  mutate(Cluster = as.factor(Cluster)) %>%
  select(-Cluster_Tissues) %>% 
  left_join(., cluster_stats, by=c('Cluster')) %>%
  mutate(Label = paste(Cluster, Cluster_Tissues,sep=': ')) %>%
  mutate(Label = ifelse(sample_accession %in% center_samples, Label, ""))

tsne_50_prep %>% mutate(Group = case_when(study_accession == 'SRP012682' ~ 'GTEx',
                                          TRUE ~ Tissue),
                        Label = case_when(grepl('^0', Label) ~ '', TRUE ~ Label),
                        Group = factor(Group, levels = c('Cornea', 'ESC', 'Lens', 'Retina', 'RPE','GTEx'))) %>% 
  ggplot(aes(x=X1,y=X2)) +
  scale_shape_manual(values=c(0:20,35:50)) +
  geom_point(size=16, alpha=0.1, aes(colour=Group)) +
  geom_point(size=6, alpha=0.6, position='jitter', aes(shape=Origin)) +
  xlab('t-SNE 1') + ylab('t-SNE 2') +
  geom_label_repel(size = 7, aes(label=Label), alpha=0.7, box.padding = unit(0.3, "lines"), force = 5) + 
  theme_bw() +
  theme(text = element_text(size = 30)) +
  custom_col + 
  guides(colour = guide_legend(override.aes = list(alpha = 1)))
```


```{r tableSampleCounts, echo = F, results='asis', warning = F, fig.cap = tableSampleCounts_cap}
tableSampleCounts_cap <- tab_cap(name='tableSampleCounts', caption = 'EiaD contains a large set of diverse eye tissues, including embryonic stem cells (ESC). Eye lid and retina endothelium samples were included, but all failed to pass our QC filters.')
table <- left_join(core_tight_2019  %>% 
                     filter(study_accession != 'SRP012682') %>% 
                     group_by(Tissue) %>% 
                     summarise(`Pre QC Count` = n()),
                   
                   core_tight_2019  %>%
                     filter(study_accession != 'SRP012682', Kept == 'Kept') %>% 
                     rowwise() %>% 
                     mutate(Sub = str_split(Sub_Tissue, ' - ')[[1]][2]) %>% 
                     mutate(Sub = case_when(is.na(Sub) ~ Sub_Tissue, TRUE ~ Sub)) %>% 
                     group_by(Tissue, Sub) %>% summarise(Count = n()) %>% 
                     mutate(Sub = paste0(Sub, ' (', Count, ')')) %>% 
                     ungroup() %>% 
                     group_by(Tissue) %>% 
                     summarise(`Count` = sum(Count), `Sub-Tissue Types (Count)` = paste(sort(unique(trimws(Sub))), collapse=', ')), by = 'Tissue') %>% 
  as_tibble() %>% 
  mutate(Tissue = case_when(Tissue == 'EyeLid' ~ 'Eye lid', TRUE ~ Tissue)) %>% 
  mutate(Count = case_when(is.na(Count) ~ 0L, TRUE ~ Count))

gtex_tissues_2017 <- core_tight_2017 %>% 
  filter(study_accession == 'SRP012682') %>%
  select(Tissue, study_accession, sample_accession) %>% unique() %>% 
  group_by(Tissue, study_accession) %>% 
  summarise(Count = n()) %>% 
  ungroup() %>% 
  mutate(Tissue = paste0(Tissue, ' (', Count, ')')) %>% 
  pull(Tissue)

gtex_tissues_2019 <- core_tight_2019 %>% 
  filter(study_accession == 'SRP012682', Kept == 'Kept') %>%
  select(Tissue, study_accession, sample_accession) %>% unique() %>% 
  group_by(Tissue, study_accession) %>% 
  summarise(Count = n()) %>% 
  ungroup() %>% 
  mutate(Tissue = paste0(Tissue, ' (', Count, ')')) %>% 
  pull(Tissue)

gtex_2017 <- cbind('GTEx', '2017', core_tight_2017 %>% 
                     filter(study_accession == 'SRP012682') %>% nrow(),
                   core_tight_2017 %>% 
                     filter(study_accession == 'SRP012682') %>% nrow(),
                   paste(gtex_tissues_2017, collapse = ', ')) %>% as_tibble()

gtex_2019 <- cbind('GTEx', '2019', core_tight_2019 %>% 
                     filter(study_accession == 'SRP012682') %>% nrow(),
                   core_tight_2019 %>% 
                     filter(study_accession == 'SRP012682', Kept == 'Kept') %>% nrow(),
                   paste(gtex_tissues_2019, collapse = ', ')) %>% as_tibble()

gtex <- rbind(gtex_2017, gtex_2019)

colnames(gtex) <- c('Group', 'Version', 'Pre QC Count', 'Count', 'Tissue Types (Count)')
# Output <div>
cat("<div id=\"tbl:tableSampleCounts\">")
regulartable(table) %>% 
  width(width = c(1,1,1,2)) %>% 
  flextable::align(align = "center",part = "header") %>% 
  flextable::align(align='center') %>% 
  flextable::fontsize(size = 10) %>% 
  font(fontname = 'Linux Libertine O', part = 'all')
# Output Caption (* * for italics)
cat(paste0("\n*",tableSampleCounts_cap,"*\n"))
# Output </div>
cat("</div>")
```

## The eyeIntegration web app provides interactive visual portal to all data 
The EiaD 2019 dataset is used directly by the eyeIntegration web app (https://eyeIntegration.nei.nih.gov). The web app was designed to provide a simple interface that has the same general concept – select specific genes and tissue and view relevant information. The web-app is divided into four general categories: expression, two-dimension sample relationships, gene networks, and data tables. 

## Custom gene and tissue expression boxplots
The 'Expression' tab of the webpage provides a wealth of information about both gene- and transcript-level expression for eye and body tissues, giving the user the ability to compare the expression of different genes within a single tissue, as well as the expression of genes across multiple tissues  (`r paste0(fig_cap(name = 'appScreenshots', display = 'cite'),'A')`). The user first selects either the 2017 or 2019 gene or transcript EiaD dataset, then Hugo Gene Nomeclature Committee (HGNC) genes names (or ENSEMBL transcripts), then tissues. A boxplot is then generated after hitting the "Re(Draw) Plot" button with overlaid individual data points. On mouse-over, the metadata for the individual sample is displayed. A tabular report is generated based on selected genes and tissues: a table with links to Ensembl, GeneCard, and OMIM for each gene for quick referencing, and a table containing expression levels for each selected gene in each selected tissue. The tables can be arranged or sorted to the user’s preference and can be easily downloaded for local use. 

```{r appScreenshots, echo = F, fig.cap = appScreenshots_cap, fig.width=12}
appScreenshots_cap <- fig_cap(name='appScreenshots', caption = 'Screenshots from eyeIntegration web app. A. Pan-tissue gene expression box plots with accompanying data tables. The data tables display the rank (lower is more highly expressed) of each gene in each sub tissue, the decile of the rank (10 is the highest decile of expression), and the mean gene\'s log2(TPM + 1) score for each sub tissue. B. Heatmap visualization.')

knitr::include_graphics('figures_and_tables/Figure 2.png')
```

Heatmap built by the R package ComplexHeatmaps based on expression can be drawn for selected genes and tissues and gene expression can be compared across many genes and tissues (`r paste0(fig_cap(name = 'appScreenshots', display = 'cite'),'B')`) [@guComplexHeatmapsReveal2016]. Finally, a session can be saved or shared by building a custom link for the session with the "Build URL Shortcut" button. 

## Differential expression and gene ontology enrichment tests allow quick comparison of gene differences between groups

We performed multiple differential comparisons at the sub-tissue level within all eye tissues and against a pan-body synthetic set comprised of a stratified sample of all tissues present in our subset of the GTEx dataset, allowing quick identification of eye specific genes across `r limma_DE_tests %>% nrow()` different comparisons. We have expanded the differential tests in the 2019 EiaD by adding the GTEx tissues as direct comparisons to our eye sub-tissues. The user can view the results selecting ‘Differential’ under the ‘Expression’ tab (`r paste0(supFig_cap(name = 'appScreenshots_sup', display = 'cite'),'D')`).  As with 'Expression', the user can select which version of the web app to draw data from as well as select for gene- or transcript-level comparisons. The user additionally has the option to select different gene classes to examine, e.g. protein coding, lincRNA.

The results of differential expression are presented in a tabular format showing log~2~ fold change, average expression, and p-values. Depending on the comparison, there are `r q01_limma_gene_sig %>% pull(Count) %>% sort() %>% head(1)` to `r q01_limma_gene_sig %>% pull(Count) %>% sort() %>% tail(1)` differentially expressed genes (Supplementary Files). The table can be easily searched for any given gene, viewed and ordered to the user’s preference, and downloaded in CSV format. Differential expression can be visualized through fold change bar graphs with the ‘Pan-tissue plots’ selection under ‘Expression’. Additionally, we performed GO enrichment for all differential comparisons. Enriched GO terms are presented first as a word cloud, for quick comparison of GO enrichment.  We provide tables, with similar viewing options as the differential expression table, for enriched GO terms in each class of a given differential comparison.

## Murine scRNA-seq enables testing of retina cell type specific expression
We incorporated scRNA-seq data from murine retina across two studies [@clarkSingleCellRNASeqAnalysis2019; @macoskoHighlyParallelGenomewide2015a]. This allows researchers to quickly examine gene expression across individual cell types in the retina. Single cell gene expression data is visualized through a heatmap showing the expression of a gene across multiple retinal cell types and different developmental time points, from embryonic day (E)11 to postnatal day (P)14 (when available), and a table of expression values is generated containing the expression data used to draw the heatmap (`r paste0(fig_cap(name = 'appScreenshots', display = 'cite'),'C')`). We also provide t-SNE/UMAP based clustering using cell type specific labeling created by the publishing authors (`r paste0(fig_cap(name = 'appScreenshots', display = 'cite'),'D, see Methods')`). The plots show all cell types present at a given developmental stage, and highlights cells expressing a gene above a user-selected given level. 

## EiaD 2019 suggests that iPSC-derived organoids and fetal retina have closely related transcriptomes
There are, currently, two major approaches to studying developing human retina: post-mortem fetal tissue and stem-cell derived organoids. We looked at how well these approaches to studying developing retina compare at a transcriptomic level, both for tissue - organoid relationships and how well they correlate across early development. 

To evaluate how the tissues and organoids compare at a transcriptome level, we looked at the same t-SNE plot from `r fig_cap(name = 'tSNE', display = 'cite')` and focused in on the three types of retina tissue (adult, fetal, and organoid) (`r paste0(fig_cap(name = 'tsneFetalRetina', display = 'cite'),'A')`). Here we saw three distinct groupings: adult retina (1), developing fetal retina and stem cell-derived organoid (2), and undifferentiated and early differentiating stem cells (3). We identified several organoid samples in cluster 3, but these share one important difference from the rest of the organoid samples in cluster 3: they have been differentiating for less than 30 days (shape 'X'). All of the organoid retina samples in cluster 2 are older than 50 days. 

To assess how similarly the fetal and organoid retina develop through time, we plotted expression of retinal progenitors, photoreceptors, and retinal ganglion markers by time in days (`r paste0(fig_cap(name = 'tsneFetalRetina', display = 'cite'),'B')`). Each row is a gene marker of either retinal progenitor, photoreceptor, or RGC. The rows are hierarchially clustered to put more similar expression patterns closer together, as denoted by the height of the dendrogram. We split the organoid tissues into three groups: Kaewkhaw et al. GFP+ and GFP- samples, and Eldred et al. samples [@eldredThyroidHormoneSignaling2018; @kaewkhawTranscriptomeDynamicsDeveloping2015]. The Kaewkhaw samples are flow sorted for a GFP marker (GFP+) under the control of the *CRX* promoter, an important regulator of photoreceptor development. GFP+ cells wouldbe enriched in photoreceptor populations. We saw that the retinal progenitor, photoreceptor, and RGC groups are largely clustered together, with patterns of expression consistent across the fetal retinal and organoid groups. 

```{r tsneFetalRetina, fig.height = 4, fig.cap=tsneFetalRetina_cap, echo=F, warning = F}
tsneFetalRetina_cap <- fig_cap(name='tsneFetalRetina', caption = 'Organoid retina, stem cell retina, and fetal retina tissue have highly similar transcriptomes. The zoom inset (A) shows the retina samples. The "Sub-Tissue Cluster" shading shows the cluster membership of the three major groups. The shapes of the points show the different origin types - notable types include the square for adult, the \'X\' for organoid under 30 days of differentiation, and the diamond for organoid over 30 days of differentiation. Major markers of retina progenitor, photoreceptors (cone and rod), and retinal ganglia cells (RGC) have similar gene expression patterns across development in retina fetal tissue and organoids.')

knitr::include_graphics('figures_and_tables/zoom_heatmap_retina.png')
```

## Differential gene expression of organoid retina versus fetal tissue identifies sets of genes relating to patterning (HOXB family), cell adhesion (protocadherin family), and RGC identity (BRN3/POU4F, NEFL, GAP43, SNCG)

To identify specific changes between retinal organoid and fetal retina tissue, we performed differential gene expression and GO term enrichment analyses. The GO term enrichment identified cell adhesion (protocadherins) and patterning (HOXB family) as enriched gene sets in retinal organoids As there is some evidence suggesting that protocadherins influence RGC viability and we noticed that several RGC markers appeared to have lower expression in the organoids compared to the fetal tissue `r paste0(fig_cap(name = 'tsneFetalRetina', display = 'cite'),'B')` we looked more closely into RGC marker expression [@lefebvreGProtocadherinsRegulateNeuronal2008].

We plotted HOXB family, protocadherin family and RGC genes in a heatmap visualization, with columns as age in days of fetal or organoid retina. Rows are genes, split by the three different groups of genes and are internally clustered by how similar the expression patterns are. We observed that there are strong, consistent gene expression differences in these three groups of genes between fetal retina and the organoid samples (`r paste0(supFig_cap(name = 'rgc_cadherin_hoxb', display = 'cite'))`). We also plotted the differential expression values between all organoids and all fetal retina samples; all genes across all three groups are significantly differentially expressed with an FDR corrected p value < 0.01.

## Data accessibility 
Individual data files for gene expression and sample metadata can be downloaded from the ‘Data’ tab on the web app. All data and code used to generate the web app can be installed from the R command line by running devtools::install_github(‘davidmcg/eyeIntegration_app’). The code for the EiaD data processing pipeline can be found at https://github.com/davidmcg/Eyeintegration_autobuild. 


# Discussion

EiaD 2019 contains a large set of carefully curated, reproducibly processed human eye RNA-seq datasets alongside a human body tissue comparison set from the GTEx project. It is available for local install as an R package at https://www.github.com/davemcg/eyeIntegration_app and it is served via a web app, eyeIntegration at https://eyeIntegration.nei.nih.gov. The web app offers a wide range of user-driven visualizations to compare expression of genes across dozens of human body and eye tissues. Furthermore, murine scRNA-seq datasets have been incorporated, allowing for examination of retina cell type-specific gene expression. Several human and non-human primate studies have been posted in the past year on the pre-print server bioRxiv and as the raw data becomes publicly available, we will be updating this section of eyeIntegration[@lukowskiGenerationHumanNeural2018; @luSinglecellRNAseqAnalysis2018; @pengMolecularClassificationComparative2018]. 

If you wish to have your data added to EiaD in the future, we suggest you 1. deposit data into GEO/SRA, 2. use clear, descriptive, consistent, and detailed metadata for each sample, and 3. (optional) contact the corresponding author. Contacting the corresponding author is only necessary if you feel your data should be included in EiaD and was deposited into the SRA before May 8th, 2019. 

As human fetal tissue is difficult to obtain and thus not very amenable for chemical or genetic modification, it is crucial for organoid-based models to be developed. Our merging of these datasets and analysis at the transcriptome level (as compared to cross-analyzing using a limited number of known marker genes) indicates that these two approaches successfully recapitulate fetal retina tissue, to a first approximation, at the whole transcriptome level. However as organoids do not develop to full function, it is important to look at how gene expression differs between retinal organoid and fetal tissue so as to suggest areas for improvement. 

We used our large dataset to narrow in on three core processes which differ significantly and substantially between retinal organoids and fetal retina. First we showed that the HOXB family is overexpressed in the organoids. The homeobox family is well known to initiate polarity of the embryo during early development [@philippidouHoxGenesChoreographers2013]. Retinoic acid is applied at about day 20 in culture to help differentiate stem cell to organoids and is also known to activate genes members of the HOXB family. The lack of HOXB expression at any age in fetal retina and the broad chromatin and gene expression changes HOXB family members can mediate suggests that HOXB activity may be unwanted for organoid maturation.

Next, we detected several protocadherins more highly expressed in the fetal tissue, relative to the organoids. Protocadherins mediate cell to cell connections and, in the developing mouse, are shown to be important for spinal internneurons and RGC survival [@hayashiEmergingRolesProtocadherins2015; @lefebvreGProtocadherinsRegulateNeuronal2008]. We would predict that decreased protocadherin expression reduces the number and maturation of RGC. Indeed we observed that many canonical RGC markers, while present in detectable levels in the organoids, are signficantly underexpressed relative to fetal tissue. This result suggests that modifying culture conditions to promote protocadherin expression may result in higher RGC yield and survival.

We built the Eye in a Disk dataset and the accompanying web app, eyeIntegration in the hopes that easily accessible gene expression across tissue space and time will be a useful tool for hypothesis generation and refinement in eye research. Wrapping all of the data processing steps in a Snakemake pipeline has several important advantages for the community: our code is publicly available for review, our analyses are reproducible, future sample updates can be streamlined in with less effort, and because all the processing is in modular pieces it is easier to add new analysis steps. In the future, we plan on regularly adding new samples to EiaD, offering *de novo* eye tissue transcriptomes, expanding the single cell RNA-seq expression tooling, adding non-human eye samples, and epigenetic datasets. 

# Supplementary Figures and Tables


```{r visitorMap, echo = F, results='asis', fig.pos = 'H', fig.cap = visitorMap_cap, fig.height=4, fig.width=10, warning=F}

ggplot(data = geo_locate, aes(x = longitudes, y = latitudes, colour = countries)) + 
  geom_map(data=world, map=world,
           aes(x=long, y=lat, map_id=region),
           color="#090D2A", fill="lightgray", size=0.05, alpha=1/4) + 
  geom_point() + 
  theme_void() +
  scale_color_viridis_d()

```

```{r mapping_rate, echo = F, message = F,  warning=F, fig.height = 4, fig.width = 10, fig.pos='H', fig.cap = mapping_rate_cap}
mapping_rate <- read_delim('data/mapping_rate.txt', delim = ' ', col_names = F) %>% 
  mutate(X2 = str_sub(X2, 1, 4) %>% as.numeric()) %>%  
  rename(sample_accession = X1, `Mapping Rate` = X2)
mapping_stats <- core_tight_2019 %>% left_join(mapping_rate) %>% pull(`Mapping Rate`) %>% summary() %>% broom::tidy()
mapping_rate_cap <- supFig_cap(name='mapping_rate_cap', caption = paste0('Salmon mapping rate for each sample, grouped by tissue type. 1st quartile mapping rate is ', round(mapping_stats$q1, 1), '%, median is ', round(mapping_stats$median,1), '%, mean is ', round(mapping_stats$mean,1), '%, and third quartile is ', round(mapping_stats$q3,1), '%.'))
core_tight_2019 %>% 
  left_join(mapping_rate, by = 'sample_accession') %>% 
  mutate(Tissue = case_when(grepl('MGS', Sub_Tissue) ~ 'Retina MGS',
                            TRUE ~ Tissue)) %>% 
  ggplot(aes(x=Tissue, y = `Mapping Rate`)) + 
  ggbeeswarm::geom_quasirandom(size = 0.5) + 
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + coord_cartesian(ylim=c(0,100))
```

```{r tSNE_sup, echo = F, message = F, warning=F, fig.width = 10, fig.pos='H', fig.cap = tSNE_sup_cap}
tSNE_sup_cap <- supFig_cap(name='tSNE_sup', caption = 'Adult retina samples from Ratnapriya et al. (MGS 1 - 4, not-AMD is MGS 1, AMD is MGS 2 through 4) cluster independently from all other adult retina samples collected.')

knitr::include_graphics('figures_and_tables/zoom_heatmap_retina_supplemental.png')
```



```{r powerCurve, echo = F, results='asis', fig.pos = 'H', fig.cap = powerCurve_cap, warning=F}

ggplot(data = power_data, aes(x=V1, y=power)) + geom_line(size = 1.5) + theme_minimal() + 
  xlab('Number of samples in each group') + ylab('Power') + scale_x_continuous(breaks = c(20,40,60,80,100))

```

```{r snakemakeDAG, echo = F, message = F, warning=F, fig.height = 8, fig.width = 10, fig.pos='H', fig.cap = snakemakeDAG_cap}
snakemakeDAG_cap <- supFig_cap(name='snakemakeDAG', caption = 'Snakemake pipeline to create EiaD 2019 consists of small modular compute sections to ensure sample tracking through the full pipeline')

knitr::include_graphics('figures_and_tables/2019_workflow.png')
```





```{r allSamplePCA, echo = F, message = F, warning=F, fig.height = 8, fig.width = 10, fig.pos='H', fig.cap = pca_cap}
# fig.pos = 'H' puts figure here, not where knitr / pandoc think is best

pca_cap <- supFig_cap(name='allSamplePCA', caption = 'PCA plot of all samples suggests that the non-eye tissue most similar to adult retina is the brain, RPE and cornea are most similar to bone marrow, blood, and skin')

# colors = viridis(n=6, direction = -1)
# colors[5] <- "red"

# pca
make_pca_tissue <- function(tissue = 'Brain'){
  ggplot(dimRed_plot_prep_pca %>% 
           mutate(Origin = case_when(study_accession == 'SRP012682' ~ 'Adult Tissue',
                                     TRUE ~ as.character(Origin))) %>% 
           filter(!is.na(Origin)) %>% 
           mutate(Group = case_when(Tissue == !!(tissue) ~ Tissue,
                                    study_accession == 'SRP012682' ~ 'GTEx',
                                    TRUE ~ Tissue)) %>% 
           mutate(Group = factor(Group, levels= c('Cornea', 'ESC', 'Retina', 'RPE', !!(tissue), 'GTEx'))),
         aes(x=`Dimension 1`,y=`Dimension 2`)) +
    scale_shape_manual(values=c(0:20,35:50)) +
    #ggtitle(paste0('All Tissue PCA: Highlight ', tissue)) +
    geom_point(size=6, alpha=0.1, aes(colour=Group)) +
    geom_point(size=1, alpha=0.8, position='jitter', aes(shape=Origin)) +
    xlab('PC 1') + ylab('PC 2') +
    theme_bw() +
    # theme(text = element_text(family = 'Linux Libertine O', size = 10)) +
    theme_minimal() +
    custom_col + 
    guides(colour = guide_legend(override.aes = list(alpha = 1)))
}
theme_set(theme_cowplot(font_size=10)) 
plot_grid(make_pca_tissue('Brain'), make_pca_tissue('Bone Marrow'),
          make_pca_tissue('Blood'), make_pca_tissue('Skin'),
          labels = c('A: Brain','B: Bone Marrow','C: Blood','D: Skin'),
          align = 'hv')
```


```{r appScreenshots_sup, echo = F, message = F, warning=F, fig.height = 8, fig.width = 10, fig.pos='H', fig.cap = appScreenshots_sup}
appScreenshots_sup <- supFig_cap(name='appScreenshots_sup', caption = 'A. CRX mouse retina gene expression heatmap and table information from Clark et al. E11 to P14 scRNA-seq. B. t-SNE visualization of gene expression profiles of individual cells from Macosko et al. C. Data table export view. D. Differential gene expression across different tissues and with GO term enrichment.')

knitr::include_graphics('figures_and_tables/eyeIntegration_web_view_01.png')
```

```{r rgc_cadherin_hoxb, echo = F, message = F, warning=F, fig.width = 10, fig.pos='H', fig.cap = rgc_cadherin_hoxb_cap}
rgc_cadherin_hoxb_cap <- supFig_cap(name='rgc_cadherin_hoxb', caption = 'Heatmap of three sets of genes across fetal retina and organoids divided by age in days (A). Bar plot of differential expression from retina to organoid, where positive values are genes that are higher expressed in fetal tissue than organoid (B). All logFC expression values have FDR corrected p value < 0.01.')

knitr::include_graphics('figures_and_tables/rgc_cadheren_box_fig.png')
```

```{r tableSampleCountsGTEx, echo = F, results='asis', warning = F, fig.cap = tableSampleCountsGTEx_cap, fig.pos = 'H'}
tableSampleCountsGTEx_cap <- supTab_cap(name='tableSampleCountsGTEx', caption = 'EiaD holds hundreds of GTEx tissues to provide a broad comparison set')
# Output <div>
cat("<div id=\"tbl:tableSampleCountsGTEx\">")
regulartable(gtex) %>% 
  width(width = c(0.9,0.9, 1,0.9,3)) %>% 
  flextable::align(align = "center", part = 'header') %>% 
  flextable::align(align='center') %>% 
  flextable::fontsize(size =8) %>% 
  font(fontname = 'Linux Libertine O', part = 'all')
#Output Caption (* * for italics)
cat(paste0("\n*",tableSampleCountsGTEx_cap,"*\n"))
# Output </div>
cat("</div>")
```

```{r tx_and_gene_types, echo = F, results='asis', warning = F, fig.cap =  tx_and_gene_types_cap, fig.pos = 'H'}
tx_and_gene_types_cap <- supTab_cap(name='tx_and_gene_types', caption = 'Dozens of different types of gene and transcript types quantified')
# Output <div>
info <- readxl::read_xlsx('data/gencode_type_description.xlsx') %>% fill('Definition') %>% 
  mutate(Biotype = case_when(Biotype == 'antisense/antisense_RNA' ~ 'antisense',
                             TRUE ~ Biotype))
cat("<div id=\"tbl:tx_and_gene_types\">")
table <- bind_rows(
  tx_anno %>% group_by(transcript_type) %>% summarise(Count=n()) %>% mutate(Type = 'Transcript') %>% rename(Class = transcript_type), 
  gene_anno %>% group_by(gene_type) %>% summarise(Count=n()) %>% mutate(Type = 'Gene') %>% rename(Class = gene_type))
table <- table %>% 
  filter(!is.na(Class)) %>% 
  rename(Biotype = Class) %>% 
  spread(Type, Count) %>% 
  rename(`Gene Count` = Gene, `Transcript Count` = Transcript) %>% 
  left_join(info, by = 'Biotype')
regulartable(table %>% arrange(Definition)) %>% 
  width(width = c(2,0.8,1,4)) %>% 
  flextable::align(align = "center", part = 'header') %>% 
  flextable::align(align='right') %>% 
   flextable::fontsize(size =6) %>% 
  font(fontname = 'Linux Libertine O', part = 'all')
#Output Caption (* * for italics)
cat(paste0("\n*",tx_and_gene_types_cap,"*\n"))
# Output </div>
cat("</div>")
```


```{r metadataTable, echo = F,  results='asis', fig.pos = 'H', fig.cap = metadata_cap}
metadata_cap <- supTab_cap(name='metadataTable', caption = 'Full metadata for 10 random eye samples. Full metadata available as supplementary file "metadata.csv."')
set.seed(1534)
cat("<div id=\"tbl:metadataTable\">")
table_ct_2019 <- core_tight_2019
colnames(table_ct_2019) <- gsub('_', ' ', colnames(table_ct_2019))
flextable(table_ct_2019 %>% filter(`study accession` != 'SRP012682') %>% sample_n(10)) %>% 
  #left_join(mapping_rate, by = 'sample_accession') %>% 
  #        rename(mapping_rate = `Mapping Rate`)) %>% 
  font(fontname = 'Linux Libertine O', part = 'all') %>% 
  flextable::fontsize(size = 5) %>% 
   flextable::fontsize(size = 5, part = 'header') %>% 
  width(width = c(0.5,0.4,0.4,0.4,0.3,0.3,0.6,1,3,0.45,0.4)) %>% 
  flextable::height_all(height = 0.1)
#Output Caption (* * for italics)
cat(paste0("\n*",metadata_cap,"*\n"))
# Output </div>
cat("</div>")
```

```{r tSNE_table, echo = F, results='asis', warning = F, fig.cap = tSNE_table_cap}
# Output <div>
cat("<div id=\"tbl:tSNE_table\">")
tSNE_table_cap <- supTab_cap(name = 'tSNE_table', caption = 'Counts of Sub Tissues in each tSNE - dbscan based cluster group')
tsne_50_prep %>% 
  select(Sub_Tissue, sample_accession, Cluster) %>% 
  mutate(`Sub Tissue` = trimws(Sub_Tissue)) %>% 
  group_by(`Sub Tissue`, Cluster) %>% 
  count(name = 'Count') %>% 
  arrange(Cluster) %>% 
  regulartable() %>% 
  width(width = c(2, 1,1)) %>% 
  flextable::align(align = "center", part = 'header') %>% 
  flextable::align(align='center') %>% 
  flextable::fontsize(size =8) %>% 
  font(fontname = 'Linux Libertine O', part = 'all')
#Output Caption (* * for italics)
cat(paste0("\n*",tSNE_table_cap,"*\n"))
# Output </div>
cat("</div>")
```






```{r GOterm_fetal_retina_organoid, echo = F, results='asis', warning = F, fig.cap = GOterm_cap, fig.pos = 'H'}
GOterm_cap <- supTab_cap(name='GOterm_fetal_retina_organoid', caption = 'Top GO terms enriched between fetal retina and organoid retina, after paring of redundant terms with REVIGO [@supekREVIGOSummarizesVisualizes2011]')
# Output <div>
cat("<div id=\"tbl:GOterm_fetal_retina_organoid\">")
organoid_fetal_GO %>% 
  dplyr::select(-ONTOLOGY, -BgRatio, -p.adjust, -Count, -Test, -Set) %>% 
  filter(ID %in% c('GO:0001503','GO:0031589','GO:0050878','GO:0061448','GO:0050678','GO:0030335','GO:0030198', 'GO:0007156', 'GO:0050769', 'GO:0030099')) %>% 
  mutate(geneID = gsub('<br>', ' ', geneID)) %>% 
  as_tibble() %>% 
  regulartable() %>% 
  flextable::fontsize(size = 8) %>% 
  width(width = c(1,1,1,1,1,2)) %>% 
  flextable::align(align = "center", part = 'header') %>% 
  flextable::align(align='center') %>% 
  font(fontname = 'Linux Libertine O', part = 'all') 

#Output Caption (* * for italics)
cat(paste0("\n*",GOterm_cap,"*\n"))
# Output </div>
cat("</div>")
```
# Acknowledgements

We would like the thank the dozens of groups who provided the raw data required to create this project. We keep a running list of the projects and associated citations at https://github.com/davemcg/eyeIntegration_app/blob/master/inst/citations.md and strongly encourage anyone who uses EiaD and eyeIntegration to cite relevant projects. We would also like to thank Kapil Bharti, Robert Hufnagel, and Brian Brooks for their continuous set of critiques and suggestions in the development of eyeIntegration app over the past two years. Tiziana Cogliati was especially helpful in the editing of this manuscript. We also would like to thank the two anonymous reviewers for their careful reading and constructive criticisms. Finally, this work utilized the computational resources of the NIH HPC Biowulf cluster (http://hpc.nih.gov).

# Funding

This research was supported by the Intramural Research Program of the National Eye Institute, National Institutes of Health. 

# Bibliography
