---
title: 'Eye in a Disk: eyeIntegration human eye and body transcriptome database version 1.00'
author:
  - Vinay Swamy:
      institute: ogvfb
  - David McGaughey:
      institute:
        - ogvfb
      correspondence: "yes"
      email: mcgaugheyd@mail.nih.gov
institute:
  - ogvfb: Ophthalmic Genetics and Visual Function Branch, National Eye Institute, National Institutes of Health
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
  word_document:
    reference_docx: word-styles-reference-01.docx
    pandoc_args:
      - '--lua-filter=scholarly-metadata.lua'
      - '--lua-filter=author-info-blocks.lua'
bibliography: eyeIntegration_v1_app_manuscript.bib
---

# Abstract

# Introduction

  From the anterior to the posterior, the human eye is composed of the cornea, lens, retina, RPE, and choroid. The development and function of these tissues is mediated through tissue and temporal specific transcript and gene expression patterns, or transcriptome. Today, RNA-sequencing (RNA-seq) is the predominant technology in quantification of the transcriptome. Analysis of gene patterns across tissue, time, and perturbation allows researchers to decipher the genetic controls of eye development and function. To this end researchers have used a wide variety of tissue types to assess gene function, including primary tissue (fetal and post-mortem), induced pluripotent stem cell (iPSC) derived cells, immortalized cell lines, and most recently, organoids. These tissue types have been deeply sequenced across the cornea [@chenIdentificationNovelMolecular2013a; @chngHighThroughputGene2013a; @chungTranscriptomicProfilingPosterior2017; @fraustoTranscriptomicAnalysisCultured2016a; @kabzaCollagenSynthesisDisruption2017; @ouyangWNT7APAX6Define2014a; @songDirectedDifferentiationHuman2016a], lens [@hanWnt5aContributesDifferentiation2018], retina [@aldiriDynamicEpigeneticLandscape2017a; @farkasTranscriptomeAnalysesHuman2013b; @hoshinoMolecularAnatomyDeveloping2017; @kaewkhawTranscriptomeDynamicsDeveloping2015; @kaewkhawTreatmentParadigmsRetinal2016; @liComprehensiveAnalysisGene2014a; @mustafiTranscriptomeAnalysisReveals2016; @pinelliAtlasGeneExpression2016; @whitmoreTranscriptomicAnalysisNasal2014b], and RPE (choroid) [@darrowDeletionDXZ4Human2016a; @harenzaTranscriptomicProfiling392017; @huIdentificationMiRNASignatures2012b; @liComprehensiveAnalysisGene2014a; @nozawaSAFARegulatesInterphase2017; @obersteinCellularResponsesHuman2017; @pengEngineeringBloodRetinalBarrier2013; @radekeRestorationMesenchymalRetinal2015a; @sainiNicotinamideAmelioratesDisease2017; @samuelAppropriatelyDifferentiatedARPE192017; @santaguidaAneuploidyinducedCellularStresses2015b; @shaoSpontaneousGenerationNovel2017; @shihRestorationMesenchymalRPE2017; @smithRetinalPigmentEpithelial2017; @stevensonGiantinknockoutModelsReveal2017; @tresiniCoreSpliceosomeTarget2015; @whewaySiRNAbasedFunctionalGenomics2015; @whitmoreTranscriptomicAnalysisNasal2014b; @auCharacterizationLincRNAExpression2017; @huIdentificationMiRNASignatures2012b]. 
  While there is a wealth of pan-eye tissue based RNA-seq data, there has not been a way for researchers to efficiently compare gene and transcript expression information across these tissues. For most human tissues the Gene Tissue Expression has generated RNA-sesq data across 
  
  
  By examining gene expression in this concerted manner researchers glean new insight into the complex regulatory mechanisms that affect gene expression and better understand the effect disease has on these mechanisms. For example, researchers used RNA-seq to establish the effects of the gene NRL on the transcriptomic regulatory network of developing photoreceptors. (Swaroop paper). Some studies have generated large RNA-seq data sets to accurately determine gene expression specific to individual tissues in the body. However, such studies require enormous amounts of funding, due to the large cost of RNA-seq and large number of samples required for these studies. Some major institutions like the NIH have performed such studies and made the data publicly available. For example, the GTEX project generated a large amount of RNA-seq data in order to determine gene expression ins 55 Human tissues and published the results in an online web-app [@gtexconsortiumGenotypeTissueExpressionGTEx2013]. This data serves as a valuable tool for researchers across the world, but lacks relevance to eye research as GTEX failed to cover the eye. 
  Another solution to the problem of creating large, high quality RNA-seq data sets is manual curation of data generated by multiple studies. There is a vast amount of publicly available RNA-seq data in the Sequence Read Archive (SRA), where investigators across the world deposit their raw sequencing data. As of January 17th 2019, the SRA has 493,821 human RNA-seq samples. Creating a curated dataset from multiple data sets presents a difficult challenge. Samples must be carefully selected and labelled with the proper metadata, evaluated for quality, and normalized to account for technical variation between studies. While many fields of research have made such databases (Other curated database), a robust database does not exist in vision science research. Previous attempts to make such a database relied on outdated technology like microarrays (iSyte), or had too narrow of focus (TIGEM).  To remedy this problem, we have created Eyeintegration, the largest curated RNA-seq database for ocular research. We provide accurate information about gene expression in multiple human eye tissues using published studies from the SRA. Additionally, we incorporate murine single cell RNA-seq data to provide researchers access to information about retinal development. The database is built using a highly reproducible and robust pipeline that includes a rigorous quality control procedure and automated publishing of the website. The database contains both gene and transcript expression data for the retina, retinal pigmented epithelium (RPE), cornea, lens and 44 other human organs. We performed multiple differential expression analysis between tissues types and provide a wide range of data visualizations that can be accessed through an easy-to-use web-app.  The database can be found at Eyeintegration.nei.nih.gov


# Methods
## We find X potential eye studies
  We searched the Sequence Read Archive^(SRA) for eye related tissues using the query ‘cornea|retina|RPE|ocular|eye’. We hand selected relevant studies and selected healthy, control/unmodified samples spanning primary adult tissue, primary fetal tissue, iPSC derived tissue, organoids and cell lines. In order to compare gene expression in the eye against expression in other body tissues, we obtained samples for 44 different body tissues from the GTEx project, also housed in the SRA. Using SRA metadata from each study we extracted sample and run accessions, library type , tissue of origin, and subtissue of origin. Any of the preceding information missing from the SRA metadata was added by hand. Stem cell derived tissues and cell lines are marked as sub-tissues of the tissue they model.   
## Data Gathering 
  We downloaded the relevant SRA files for each sample directly from the NCBI ftp server using the file transfer software Aspera. SRA files were converted to FASTQ format using the tool fastq-dump from the SRAtoolkit software package^. Samples only available in the BAM format were converted to FASTQ format using SAMTOOLS^. Samples were quantified using the alignment free quantification software Salmon, using transcriptomic index built from  gencode v28 protein coding transcript sequences^ using the transcriptomic aligner Salmon^. Using the resulting expression data, we identified lowly or unused transcripts within annotation, and removed transcripts that accounted for 5% or less of the total expression for its parent gene.^(sonneson et al). Samples were requantified against a transcriptomic index built on the filtered transcript sequences. All expression values in length-scaled TPM counts were combined into a single matrix and aggregated to the gene level using tximport^.  
 
## Quality control 
We began the quality control procedure by removing samples with a Salmon mapping rate less than 50%. We further filtered samples by calculating the median gene expression for each sample and only kept sample that had a median count >2 across all genes. We removed lowly expressed genes by calculating the median expression across all samples for each gene and kept genes that had a median count>200 across all samples. To reduce the noise experimental variability between each study, we normalized samples by library using the calcNormFactors function from the edgeR r-package, and then quantile smoothed expression data using the R package qsmooth^(Hicks et al).  
We generated sub-tissue clusters by first reduced the dimensionality of the data to two dimensions using the t-Stochastic Neighbor Embedding(t-SNE) algorithm, and then calculated the center of the resulting sub-tissue clusters by finding the average coordinate of all samples  of a given sub-tissue type. We used these sub-tissues clusters to identify outlier samples, or samples that failed to cluster with other samples of the same sub-tissue type. We calculated the distance of each sample from its sub-tissue cluster center and removed samples that had distance 4 standard deviations from the average distance. 
 
 
## Analysis and generation of web app 
We used the cleaned data to determine differential gene expression between different sub-tissue types. First, we generated a synthetic body set to serve as single representative sub-tissue type for pan-body gene expression by randomly sampling GTEx tissues.  We used the voom function from the Limma R package to convert gene expression to precision weights, and then performed pairwise differential expression tests for all combinations of sub-tissues using an empirical Bayes test. We extracted significant genes (p <.05) for each eye sub-tissue vs body comparisons and used these to calculate GO enrichment for each eye sub-tissue. The significant gene list for each eye sub-tissue was split into upregulated and down regulated sets and each set was tested for enrichment using the enrichGO function from the clusterProfiler  R-package (q-value < .01).  
 
VS: 
I’ll leave the webapp and single cell stuff for you since I didn’t really work on that part 
# Results

## Data collection and Quality control 
Our initial query to the SRA found 44* potentially relevant studies. We removed non-pertinent studies and selected healthy or unmodified tissue from each relevant study for a total of <x> studies and 410 samples, with 196 Retina, 143 RPE, 62 Cornea, and 9 lens samples. We built an automated pipeline for processing and analyzing all data for the web app using the program Snakemake^, a workflow management system that allows for efficient parallel execution of the analysis, facilitates reproduction by others, and simplifies long term maintenance of the web-app. In order to create a robust and clean data set, we developed a rigorous quality control procedure as part of our analysis, considering a sample’s read mapping rate and median count level as well as behavior relative to samples of the same tissue type.  We used t-SNE to cluster samples passing quality controls by tissue type to observe variability within samples of the same tissue and determine the strength of our quality control procedure. (figure 1A) We found that samples of the same tissue type and origin cluster well together. For example, in retina derived tissue, primary adult tissue clusters tightly and distinctly from other cell types, and retinal organoids and fetal tissue cluster together. Our ability to uniformly cluster data by known biological function independent of study origin demonstrates our quality control method’s ability to account for technical variation between studies. We provide plots of t-SNE clusters at variable perplexities within the web-app under the ‘2D-clustering’ tab. After quality control we found 310 samples to be of sufficient quality, with 180 Retina, 117 RPE, 42 Cornea, and 6 Lens samples included in downstream analysis. 
The 2019 version of the web-app represents a significant change from our previously published Eyeintegration 2017 data set. (Table 1a/Figure 1b) We have added 80 more Retina samples, 61 more RPE samples, 22 more cornea samples, and added Lens as a new tissue type, across <x> more studies. We have increased the number of sub-tissue types, adding Retinal fetal tissue, retinal organoids and cornea stem cell lines. (Fig. 1A). There has been significant change to the types of gene classes. We have added miRNA, lincRNA, pseudogenes, and snRNA as gene classes. The web app contains 151222 protein coding, 13188 lincRNA, 10202 pseudogenes, 1900 snRNA and 1881 miRNA transcripts, for a net addition of X more genes and Y more transcripts.  (Table 1b?).  
 
## eyeIntegration web application provides interactive visual portal to all data 
All data generated by our analysis is presented through the Eyeintegration web-app. The web-app was designed to provide a simple interface that have the same general concept – select specific genes and tissue and view relevant information. The web-app is divided into 3 general categories: gene/transcript expression, differential comparisons, and single cell data. 
 
## Expression 
The expression tab of the webpage provides a wealth of information about both gene and transcript expression in a range of eye and body tissues, giving the user the ability to compare the expression of different genes within a single tissue, as well as the expression of genes across multiple tissues. (Fig 2) The user first selects which version of the web-app to use data from, as well as genes and tissues to examine. A tabular report is generated based on selected genes and tissues: a table with links to Ensembl, GeneCard, and OMIM for each gene for quick referencing, and a table containing expression levels for each selected gene in each selected tissue. The tables can be arranged or sorted to the user’s preference, and can be easily downloaded for local use. The web app provides multiple options for visualizing expression data. Box plots can be drawn to compare gene expression across tissues.  Users may select any number of genes and tissues to compare at a given time, and can arrange visualizations in multiple ways. Box plots provide the user a measure of confidence in the variability of reported gene expression levels. The metadata behind any given samples can be seen by hovering over a point in the box plot, including study of origin, sex of sample, and SRA accession.  Heatmaps based on expression can be drawn for selected genes and tissues, and are useful for comparing gene expression across many genes and tissues. The visualizations can also be generated at the transcript level by selecting pan-transcript expression under the expression tab. Finally, a session can be saved or shared by building a custom link for the session with the build URL shortcut button. 
 
## Differential expression and Gene Ontology data. 
Using our expression data, we performed multiple differential comparisons at the sub-tissue level. We performed differential tests of eye tissues against a pan-body synthetic set comprised of a stratified sample of all tissues present in the GTeX data set, allowing users to quickly identify eye specific genes. We also performed differential tests between individual eye tissues. The user can view the results under the ‘differential’ selection under the ‘expression’ tab. (fig 3) As with expression, the user can select which version of the web app to draw data from as well as select for gene or transcript level comparisons. The user additionally has the option to select different gene classes to examine, protein coding, lincRNA etc. The results of differential expression are presented in a tabular format showing log2 fold change, average expression, and different statistical values. The table can be easily searched for any given gene, viewed and ordered to the user’s preference, and downloaded in CSV format. Differential expression can be visualized through fold change bar graphs in the ‘pan-tissue plots’ section under expression. Additionally, we performed GO enrichment for all differential comparisons. Enriched GO terms are presented first as a word cloud, for quick comparison of GO enrichment.  We provide tables, with similar viewing options as the differential expression table, for enriched GO terms in each class of a given differential comparison 
 
## Single Cell data 
Single Cell RNA-seq provides a deeper look into the   
We provide single cell RNA-seq data from murine retina across 2 studies. The data is taken directly from the publishing authors. Single cell gene expression data is visualized through a heatmap showing the expression of a single gene across multiple retinal cell types and different developmental time points (E11-P14), and a table of expression values is generated containing the expression data used to draw the heatmap. (fig?) We also provide t-SNE/UMAP based clustering using cell type specific labeling created by the publishing authors. (fig?) The plots show all cell types present a given developmental stage, and highlights cells expressing a gene at a user determined level. 
 
## Data accessibility 
Individual data files for gene expression and sample metadata can be downloaded from the ‘Data’ tab. All data and code used to generate the web app can be installed from the R command line by running devtools::install_github(‘davidmcg/Eyeintegration_app’). Code used to generate data processing pipeline can be found at https://github.com/davidmcg/Eyeintegration_autobuild. 
 
Fig 1A – bulk tissue tsne 
Fig 1b new samples 
Table 1a counts at 2019/2017 
Table 1b #tx/genes for each gene class at different times 2019/2017 
Fig2 screen shots of expression 
Fig3 screen shots of differential comparisons 
(sup)Fig 4 screen shot of single cell data. 
# Conclusions

# Bibliography
